niap-cc/Permissions/Tester/app/src/main/java/com/android/certifications/niap/permissions/InstallPermissionTester.java

```diff
@@ -177,35 +177,52 @@ public InstallPermissionTester(TestConfiguration configuration, Activity activit
 
         mPermissionTasks.put(ACCESS_WIFI_STATE,
-                new PermissionTest(false, () -> mWifiManager.getConfiguredNetworks()));
+                new PermissionTest(false, () -> {
+                    if (ActivityCompat.checkSelfPermission(mContext, Manifest.permission.ACCESS_FINE_LOCATION)
+                            != PackageManager.PERMISSION_GRANTED) {
+                        throw new BypassTestException(
+                                "ACCESS_FINLE_LOCATION permission should be granted to run this test case");
+                    }
+                    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
+                        //mWifiManager.getCallerConfiguredNetworks();
+                        mWifiManager.getConfiguredNetworks();
+                    } else {
+                        mWifiManager.getConfiguredNetworks();
+                    }
+                }));
 
         // android.permission.AUTHENTICATE_ACCOUNTS has been removed.
 
         // Starting in Android 12 the install BLUETOOTH permission is no longer used and has instead
         // been replaced by the new runtime BLUETOOTH permissions.
         mPermissionTasks.put(BLUETOOTH,
                 new PermissionTest(false, Build.VERSION_CODES.P, Build.VERSION_CODES.R, () -> {
-            if (mBluetoothAdapter == null) {
-                throw new BypassTestException(
-                        "A bluetooth adapter is not available to run this test");
-            }
-            mBluetoothAdapter.getAddress();
-        }));
+                    if (mBluetoothAdapter == null) {
+                        throw new BypassTestException(
+                                "A bluetooth adapter is not available to run this test");
+                    }
+                    mBluetoothAdapter.getAddress();
+                }));
 
         // Starting in Android 12 the BLUETOOTH_ADMIN permission is no longer used and has instead
         // been replaced by the new runtime BLUETOOTH permissions.
         mPermissionTasks.put(BLUETOOTH_ADMIN,
                 new PermissionTest(false, Build.VERSION_CODES.P, Build.VERSION_CODES.R, () -> {
-            if (mBluetoothAdapter == null) {
-                throw new BypassTestException(
-                        "A bluetooth adapter is not available to run this test");
-            }
-            if (mBluetoothAdapter.isEnabled()) {
-                mBluetoothAdapter.disable();
-                mBluetoothAdapter.enable();
-            } else {
-                mBluetoothAdapter.enable();
-            }
+                    if (mBluetoothAdapter == null) {
+                        throw new BypassTestException(
+                                "A bluetooth adapter is not available to run this test");
+                    }
+                    if (mBluetoothAdapter.isEnabled()) {
+                        if (ActivityCompat.checkSelfPermission(mContext, Manifest.permission.BLUETOOTH_CONNECT)
```

Comment:
This permission check should be removed; this test has a maxSdkVersion of R, but the `BLUETOOTH_CONNECT` permission was added in S, so if bluetooth is enabled, this test will be skipped on all devices running Android P-R.

Reply:
Okay removed.

Reply:
Update looks good; thanks!